src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -214,6 +216,105 @@ export async function generateDownloadableLinks(
     }
 }
 
+export async function downloadableLinks(
+    periscopeStorage: PeriscopeStorage,
+    output: string,
+    nodeName: string
+): Promise<any> {
+
+    try {
+        const storageAccount = periscopeStorage.storageName;
+        const storageKey = periscopeStorage.storageKey;
+
+        // Get DNS Core hostname which Periscope use it as name of the container.
+        const matches = output.match(/(https?:\/\/[^\s]+)/g);
+        let containerName;
+        if (matches![0].indexOf('://') !== -1) {
+            containerName = matches![0].replace('https://', '').split('.')[0];
+        }
+
+        // Use SharedKeyCredential with storage account and account key
+        const sharedKeyCredential = new StorageSharedKeyCredential(storageAccount, storageKey);
+        const blobServiceClient = new BlobServiceClient(
+            `https://${storageAccount}.blob.core.windows.net`,
+            sharedKeyCredential
+        );
+
+        // Hide this all under single pupose funct which lik elike : getZipDir or get me dirname
+        const containerClient = blobServiceClient.getContainerClient(containerName);
+
+        // List all current blob.
+        const listCurrentUploadedFolders = [];
+        for await (const item of containerClient.listBlobsByHierarchy("/")) {
+            if (item.kind === "prefix") {
+                listCurrentUploadedFolders.push(item.name);
+            }
+        }
+
+        // Sort and get the latest uploaded folder under the container used by periscope.
+        // const periscopeHtmlInterfaceList = [];
+        // jszip for zipping
+        const zip = new JSZip();
+
+        for await (const blob of containerClient.listBlobsFlat()) {
+            // Get the latest uploaded folder, then Identify the Zip files in the latest uploaded logs within that folder
+            // and extract *.zip files which are individual node logs.
+            const latestBlobUploadedByPeriscope = blob.name.indexOf(listCurrentUploadedFolders.sort().reverse()[0]);
+
+            if (latestBlobUploadedByPeriscope !== -1 && path.extname(blob.name) !== '.zip' && blob.name.indexOf(nodeName) !== -1) {
+                // Get a block blob client
+                // https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-nodejs-legacy#download-a-blob
+                const blockBlobClient = containerClient.getBlockBlobClient(blob.name);
```

Comment:
Could everything from here to the bottom of the loop be a well-named helper function?  It seems like a `downloadBlobToFile` kind of chunk.

Reply:
Oh looking below maybe this isn't downloading the blob to a file.